src/core/stream/representation/get_needed_segments.ts

```diff
@@ -103,8 +114,32 @@ export default function getNeededSegments({
   getBufferedHistory,
   neededRange,
   segmentsBeingPushed,
+  maxVideoBufferSize,
 } : IGetNeededSegmentsArguments) : ISegment[] {
   const { representation } = content;
+  const estimateSizeBeingPushed = segmentsBeingPushed.reduce((size, segment) => {
+    const { bitrate } = segment.representation;
+    // Not taking into account the fact that the segment
+    // can still be generated and the duration not fully exact
+    const { duration } = segment.segment;
+    return size + (bitrate * duration);
+  }, 0);
+  let availableBufferSize = bufferedSegments.reduce((size, chunk) => {
+    if (chunk.chunkSize !== undefined) {
+      return size - chunk.chunkSize;
+    } else {
+      return size;
+    }
+
+  } , maxVideoBufferSize) - estimateSizeBeingPushed;
+
+  // Current buffer length in seconds
+  let bufferLength = bufferedSegments.reduce((length, segment) => {
+    return length + (segment.end - segment.start);
+  }, 0)
+  + segmentsBeingPushed.reduce((length, segment) => {
+    return length + segment.segment.duration;
+  }, 0);
```

Comment:
Could be combined with the previous logic eventually.

Also weird to have the line 140 be idented as a regular line yet beginning with a +, it's not immediately evident there's an operation in there.